adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapterWithClientTest.java

```diff
@@ -0,0 +1,414 @@
+package org.eclipse.hono.adapter.amqp;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.TimeUnit;
+
+import org.apache.qpid.proton.amqp.messaging.Rejected;
+import org.apache.qpid.proton.amqp.messaging.Released;
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.client.HonoClient;
+import org.eclipse.hono.client.MessageSender;
+import org.eclipse.hono.client.RegistrationClient;
+import org.eclipse.hono.client.ServerErrorException;
+import org.eclipse.hono.client.TenantClient;
+import org.eclipse.hono.config.ProtocolAdapterProperties;
+import org.eclipse.hono.service.command.CommandConnection;
+import org.eclipse.hono.util.RegistrationConstants;
+import org.eclipse.hono.util.TenantConstants;
+import org.eclipse.hono.util.TenantObject;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import io.vertx.proton.ProtonClient;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonHelper;
+import io.vertx.proton.ProtonQoS;
+import io.vertx.proton.ProtonSender;
+
+/**
+ * Verifies the behaviour of {@link VertxBasedAmqpProtocolAdapter}. In this setup, the AMQP adapter is deployed and run
+ * as a Verticle. A ProtonClient, which simulates a device, is used to communicate with the adapter.
+ */
+@RunWith(VertxUnitRunner.class)
+public class VertxBasedAmqpProtocolAdapterWithClientTest {
```

Comment:
These seem to be functional tests running against a deployed version of the adapter. I would rather move these tests to the `tests` module and run the tests against the Docker containers we start there. It would probably also make more sense to put it into a separate PR then...

Reply:
ok, i see. thanks!